src/basic/efivars.c

```diff
@@ -96,6 +96,17 @@ int efi_get_variable(
                                 (void) usleep_safe(EFI_RETRY_DELAY);
                 }
 
+                /* According to comment of linux source code, efivarfs represents uncommitted
+                 * variables with zero-length files. Reading them should return EOF, even though
+                 * st.st_size > 0
+                 *
+                 * See https://github.com/systemd/systemd/blob/b6b8527cd184085008585c1a1f9725eb97c342ef/src/basic/efivars.c#L99-L101
+                 */
+                if (n == 0) {
+                        return log_debug_errno(SYNTHETIC_ERRNO(ENOENT),
+                                               "EFI variable %s is uncommitted", p);
+                }
```

Comment:
Please drop the `{}` and add a comment that if the kernel changes behaviour (to flush dentries on resume), we can drop this at some point in the future. But note that https://github.com/torvalds/linux/commit/3fab70c165795431f00ddf9be8b84ddd07bd1f8f is 11 years old at this point so we'll need to deal with the current behaviour for a long time.